inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl

```diff
@@ -0,0 +1,429 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "include/include_all.cl"
+
+#define SIMD 16
+#define FSV 16
+#define unroll_for  __attribute__((opencl_unroll_hint(READ_OFFSET))) for
+
+#if !defined REDUCE_BATCH
+    #define REDUCE_BATCH 0
+#endif
+#if !defined REDUCE_FEATURE
+    #define REDUCE_FEATURE 0
+#endif
+#if !defined REDUCE_Y
+    #define REDUCE_Y 0
+#endif
+#if !defined REDUCE_X
+    #define REDUCE_X 0
+#endif
+
+#if INPUT0_TYPE_SIZE == 4
+    #define BLOCK_READ(ptr, offset) DT_INPUT_BLOCK_READ4(ptr, offset);
+    #define READ_OFFSET 4
```

Comment:
Read offset is already selected in host-code, so this is repeating the logic.
Wouldn't it make more sense to have `READ_OFFSET` as jit from host and use
```c
#define BLOCK_READ(ptr, offset) MAKE_VECTOR_TYPE(DT_INPUT_BLOCK_READ, READ_OFFSET)(ptr, offset)
```
?